Source/Core/DolphinQt/Debugger/CodeDiffDialog.cpp

```diff
@@ -133,6 +156,101 @@ void CodeDiffDialog::OnClickItem()
   m_code_widget->SetAddress(address, CodeViewWidget::SetAddressUpdate::WithDetailedUpdate);
 }
 
+void CodeDiffDialog::SaveDataBackup()
+{
+  if (Core::GetState() == Core::State::Uninitialized)
+  {
+    ModalMessageBox::information(this, tr("Code Diff Tool"),
+                                 tr("Emulation must be started before saving a file."));
+    return;
+  }
+
+  if (m_include.empty())
+    return;
+
+  std::string filename =
+      File::GetUserPath(D_LOGS_IDX) + SConfig::GetInstance().GetGameID() + "_CodeDiff.txt";
+  File::IOFile f(filename, "w");
+  if (!f)
+  {
+    ModalMessageBox::information(this, tr("Code Diff Tool"),
+                                 tr("Failed to save file to: ") + QString::fromStdString(filename));
+    return;
+  }
+
+  // Copy list of BLR tested functions:
+  std::set<u32> address_blr;
+  for (int i = 0; i < m_matching_results_table->rowCount(); i++)
+  {
+    if (m_matching_results_table->item(i, 4)->text() == QStringLiteral("X"))
+      address_blr.insert(m_matching_results_table->item(i, 4)->data(Qt::UserRole).toUInt());
+  }
+
+  for (const auto& line : m_include)
+  {
+    bool blr = address_blr.contains(line.addr);
+    f.WriteString(
+        fmt::format("{} {} {} {:d} {}\n", line.addr, line.hits, line.total_hits, blr, line.symbol));
+  }
+}
+
+void CodeDiffDialog::LoadDataBackup()
+{
+  if (Core::GetState() == Core::State::Uninitialized)
+  {
+    ModalMessageBox::information(this, tr("Code Diff Tool"),
+                                 tr("Emulation must be started before loading a file."));
+    return;
+  }
+
+  if (g_symbolDB.IsEmpty())
+  {
+    ModalMessageBox::warning(
+        this, tr("Code Diff Tool"),
+        tr("Symbol map not found.\n\nIf one does not exist, you can generate one from "
+           "the Menu bar:\nSymbols -> Generate Symbols From ->\n\tAddress | Signature "
+           "Database | RSO Modules"));
+    return;
+  }
+
+  std::string filename =
+      File::GetUserPath(D_LOGS_IDX) + SConfig::GetInstance().GetGameID() + "_CodeDiff.txt";
+  File::IOFile f(filename, "r");
+  if (!f)
+  {
+    ModalMessageBox::information(this, tr("Code Diff Tool"),
+                                 tr("Failed to find or open file: ") +
+                                     QString::fromStdString(filename));
+    return;
+  };
+
+  ClearData();
+
+  std::set<u32> blr_addresses;
+  char line[512];
+  while (fgets(line, 512, f.GetHandle()))
+  {
+    bool blr = false;
+    Diff temp;
+    std::istringstream iss(line);
+    iss.imbue(std::locale::classic());
+    iss >> temp.addr >> temp.hits >> temp.total_hits >> blr >> std::ws;
+    std::getline(iss, temp.symbol);
+    m_include.push_back(std::move(temp));
+
+    if (blr)
+      blr_addresses.insert(temp.addr);

```

Comment:
This uses `temp` after it has been moved.  I'm assuming that is not intended?

Reply:
Good catch!

Reply:
I'm surprised this didn't seem to affect the behavior :thinking:

Reply:
I think the auto-generated move constructor might be copying `addr` without overwriting the value in the old object. Because `addr` is just a u32, that would be the most efficient approach for it. But either way, we shouldn't rely on it.